### PR TITLE
Fix regression in -files command

### PR DESCRIPTION
**Pull request by insomniacslk**
This commit addresses https://github.com/u-root/u-root/issues/650

### CODE REVIEW

**Comment by insomniacslk** [comment]
This PR is not yet ready to be merged. I will make two additional commits, for a total of:

* fixing the `-files` flag regression (the first commit)
* add a CircleCI test to make sure we won't regress again
* support the globbing syntax as suggested in https://github.com/u-root/u-root/issues/651

**Comment by insomniacslk** [comment]
This commit was tested by calling `u-root` in the following ways:

* `-files /bin/bash` (works, as expected)
* `files /bin/bash:/bin/bash` (fails, as expected)
* `-files /bin/bash:bin/besh` (works, as expected)
* `-files /bin/bash -files /etc/passwd`
* `-files /bin/bash:bin/myshell -files /etc/passwd:passwd`

and verifying that the files were in the expected locations in the `.cpio` file

**Comment by insomniacslk** [comment]
@hugelgupf this should fix the regression, with the exception of globbing that I commented on the related issue. Mind having a look when you have time?

File: .circleci/config.yml [hunk 1]
```diff
@@ -245,3 +260,126 @@ jobs:
...
+          name: Install cpio
+          command: sudo apt-get install cpio
+      - run:
+          name: Build u-root
+          environment:
+            - GOARCH: amd64
+          command: go build u-root.go
+      - run:
+          name: Build ramfs
+          command: ./u-root -build=bb --tmpdir=/tmp/u-root -files /bin/bash
```
Comment (rminnich) [inline .circleci/config.yml]: change this to ///bin/bash

File: .circleci/config.yml [hunk 2]
```diff
@@ -245,3 +260,126 @@ jobs:
...
+          name: Install cpio
+          command: sudo apt-get install cpio
+      - run:
+          name: Build u-root
+          environment:
+            - GOARCH: amd64
+          command: go build u-root.go
+      - run:
+          name: Build ramfs
+          command: ./u-root -build=bb --tmpdir=/tmp/u-root -files /bin/bash -files /bin/ls
```
Comment (rminnich) [inline .circleci/config.yml]: add a relative path here
mkdir etc
-files etc

File: .circleci/config.yml [hunk 3]
```diff
@@ -245,3 +260,126 @@ jobs:
...
+          name: Install cpio
+          command: sudo apt-get install cpio
+      - run:
+          name: Build u-root
+          environment:
+            - GOARCH: amd64
+          command: go build u-root.go
+      - run:
+          name: Build ramfs
+          command: ./u-root -build=bb --tmpdir=/tmp/u-root -files /bin/bash:bin/bush
```
Comment (rminnich) [inline .circleci/config.yml]: bin/bush?

File: pkg/uroot/uroot.go [hunk 4]
```diff
@@ -177,7 +177,7 @@ func CreateInitramfs(opts Opts) error {
 		} else {
 			// plain old syntax
 			src = file
-			dst = file
+			dst = strings.TrimLeft(file, "/")
```
Comment (rminnich) [inline pkg/uroot/uroot.go]: ok, I did this differently.First off, it fails with paths like ///bin/bash. So.
src = filepath.Clean(src)
dst = src
if path.IsAbs(dst) {
dst = filepath.Rel("/", dst)
}

or some such. As is it's a bit fragile and people can break it if their scripts don't give you clean paths.

File: .circleci/config.yml (hunk 3 above)
Comment (insomniacslk) [reply .circleci/config.yml]: yes, to confirm that file with a different name from the source will work too

File: pkg/uroot/uroot.go (hunk 4 above)
Comment (insomniacslk) [reply pkg/uroot/uroot.go]: I've addressed the comments above. Checking on this one shortly

**Comment by insomniacslk** [comment]
@rminnich I should have addressed your feedback, can you please have a look again?

**Pull request merged**